packages/core/src/utils/LockManager.ts

```diff
@@ -0,0 +1,47 @@
+import { v4 as uuidv4 } from 'uuid';
+
+class LockManager {
+  private static _instance: LockManager;
+  private locks: Map<string, boolean>;
+
+  private constructor() {
+    this.locks = new Map();
```

Comment:
Curious what are your thoughts on choosing Map vs Set?

Reply:
map is more flexible and intuitive in our case, as we might need to add the status of the lock.

Reply:
> as we might need to add the status of the lock

Are there any use cases that you foresee?

Reply:
**Status of the Lock**:
The current setup serves as a sample and doesn't include this feature. However, it can be integrated if needed. An example use-case would be as follows:

```
  childProcess.stdin?.write(
    //Update lock status
    content,
    (e) => {
      if (e) {
        logger.debug(e as unknown as string);
        logger.error(`Error generating ${imageOutputPath}`);
      }
      childProcess.stdin?.end();
    },
  );

  childProcess.on('error', (error) => {
    logger.debug(error as unknown as string);
    logger.error(`Error generating ${imageOutputPath}`);
    LockManager.deleteLock(lockId);
  });

  childProcess.stderr?.on('data', (errorMsg) => {
     //Update lock status
    errorLog += errorMsg;
  });

  childProcess.on('exit', () => {
    // This goes to the log file, but not shown on the console
    logger.debug(errorLog);
    LockManager.deleteLock(lockId);
  });
```